File: src/main/java/org/apache/prepbuddy/cluster/FingerprintAlgorithm.java
package org.apache.prepbuddy.cluster;

import java.util.regex.Pattern;

public abstract class FingerprintAlgorithm implements ClusteringAlgorithm {
    private static final Pattern PUNCTUATION_MATCHER = Pattern.compile("\\p{Punct}|[\\x00-\\x08\\x0A-\\x1F\\x7F]");

    static String removeAllPunctuations(String someString) {
        return PUNCTUATION_MATCHER.matcher(someString).replaceAll("");
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/clusterers/FingerprintAlgorithm.scala
package org.apache.datacommons.prepbuddy.clusterers

import java.util.regex.Pattern

abstract class FingerprintAlgorithm extends ClusteringAlgorithm {
    private val PunctuationMatcher: Pattern = Pattern.compile("\\p{Punct}|[\\x00-\\x08\\x0A-\\x1F\\x7F]")

    def removeAllPunctuations(someString: String): String = PunctuationMatcher.matcher(someString).replaceAll("")
}
